logstash-core/lib/logstash/runner.rb

```diff
@@ -248,6 +261,10 @@ def execute
 
     return start_shell(setting("interactive"), binding) if setting("interactive")
 
+    module_parser = LogStash::ModulesCLIParser.new(@modules_list, @modules_variable_list)
+    # Now populate Setting for modules.list with our parsed array.
+    @settings.set("modules.config", module_parser.output)
+
```

Comment:
Nice, I would have expected it to be more hackery but this is a nice way to do it.